base/generator.jl

```diff
@@ -0,0 +1,21 @@
+"""
+    Generator(f, iter)
+
+Given a function `f` and an iterator `iter`, construct an iterator that yields
+the values of `f` applied to the elements of `iter`.
+The syntax `f(x) for x in iter` is syntax for constructing an instance of this
+type.
+"""
```

Comment:
I saw this as the equivalent of python's `itertools.imap`, a lazy map (except that `imap` accepts multiple iterators).